suite/tests/api/ir_riscv64.expect

```diff
@@ -221,8 +221,6 @@ c.xor  a5 -> fp
 c.sub  a5 -> fp
 test_integer_arith complete
 lui    0x2a -> a0
-<Internal Error: Failed to encode instruction: 'auipc  0x0000004000016234 -> a0'
```

Comment:
We should continue printing this error (and not crashing). Perhaps when `check_reachable` is set to true.

Reply:
This error message is printed by `SYSLOG_INTERNAL_ERROR`, so I think we could use `SYSLOG_INTERNAL_WARNING` for check_reachable printing (without crashing). But with this change:

```diff
    if (enc == ENCFAIL) {
+       if (check_reachable) {
+           char disas_instr[MAX_INSTR_DIS_SZ];
+           instr_disassemble_to_buffer(dcontext, instr, disas_instr, MAX_INSTR_DIS_SZ);
+           SYSLOG_INTERNAL_WARNING("Internal Error: Failed to encode instruction: '%s'\n", disas_instr);
+        }
```

It seems `SYSLOG_INTERNAL_WARNING` evaluated to empty in suite/tests/bin/api.ir-static, causing a difference with suite/tests/bin/api.ir. Should we just use `print_file(STDERR, ...)`?

Reply:
But if it's the debug build `SYSLOG_INTERNAL_ERROR` should be set appropriately. It was before when we were crashing right?

Reply:
> It was before when we were crashing right?

I also thought so, but it's not: https://github.com/DynamoRIO/dynamorio/actions/runs/5848221837/job/15855283218#step:6:1932. The reason CI was failing is that -- in api.ir-static, `SYSLOG_INTERNAL_ERROR` evaluated to empty while in api.ir it's not. `SYSLOG_INTERNAL_ERROR` is (surprisingly) not causing a crash, so assert_reachable is not really asserting.

And the reason why `SYSLOG_INTERNAL_ERROR` is evaluated to empty in api.ir-static is that -- api.ir-static is built as a STANDALONE_DECODER.

Reply:
> I also thought so, but it's not

So then maybe we don't need to set `assert_reachable` to `false` in this PR? Let's revert that part.

>  so assert_reachable is not really asserting.

I see #5298 added the `assert_reachable` arg. The description says that it silences a warning, not an assert statement. I guess the name `assert_reachable` could be changed to `warn_reachable` or something, but let's not change that in this PR.

>  in api.ir-static, SYSLOG_INTERNAL_ERROR evaluated to empty while in api.ir it's not.

Perhaps the `ASSERT(pc == NULL)` in `test_instr_encoding_failure` is all we need then. For api.ir where the warning will be printed we can have a regex like `(<message>)?` which would work also for api.ir-static.

Reply:
> we can have a regex like `(<message>)?`

For api.ir, we are using `runcmp.cmake` to compare which uses `diff`:

```
if (AARCH64 OR RISCV64)
  # The ir_aarch64.expect file is too large for CMake's regexes.
  set(api.ir-static_runcmp "${CMAKE_CURRENT_SOURCE_DIR}/runcmp.cmake")
  set(api.ir-static_runcmp_capture "stderr")
endif ()
```

Maybe set assert_reachable to false is the way to go?

Reply:
> Maybe set assert_reachable to false is the way to go?

This doesn't sound like a good idea now that we understand it's quite innocuous and suits non-test code well. The real problem here seems to be that api.ir and api.ir-static use the same `expect` file. I'm not suggesting we split that because they share most of their contents. I also see
```
    # The ir_aarch64.expect file is too large for CMake's regexes.
```
so we cannot just switch to using regexps.

Circling back to your earlier suggestion of using `print_file(STDERR, ...)`:

@derekbruening why do we not define syslog macros for the stand-alone case? Can we just define those macros? Or maybe replace this particular log with an explicit call to `d_r_notify`?

Reply:
I would guess it's because for STANDALONE_DECODER there is no mechanism to turn off notifications because there is no place to set stderr_mask or msgbox_mask.  Remember that on Windows an error by default has a pop-up window which is really annoying if you can't turn it off (at the same time it kind of has to default to that for GUI apps run without a console where a stderr print will never be visible).  Maybe if you set msgbox_mask to 0 for STANDALONE_DECODER and have some way to set options -- which ties into us wanting a real initialization for all decoder use which could solve it by reading the DYNAMORIO_OPTIONS env var: xref #2499, #6002.

Reply:
@ksco From offline discussion with Derek: since runcmp.cmake is our own script we can perhaps add macros or something to it so that the additional logs are expected only in the non-static test. I haven't looked at the implementation myself so you might want to check if that's easily done in this PR or not.

Reply:
Possibilities:
+ Add per-line (to avoid too-large issue) regex support in runcmp
+ Split the expect file and share the common with `#include` which either runcmp is augmented to look at or you run cpp
+ Add conditional marker capabilities: maybe leveraging how suite/tests/CMakeLists.txt does it for .template* files, or something new in runcmp

Reply:
Sure, thank you! I’ll take a look.